### ISSUE TITLE
"Panorama Background On All Screens" breaks YetAnotherConfigLib screens

### ISSUE THREAD

**Issue by SAJewers** [juancarloscp52/BedrockIfy#289]
- Enable "Panorama Background On All Screens"
- Go to the options for a mod that uses YACL (eg. Debufigy or Mixtape)
- Switch to a different tab

The config screen immediately goes blank, and you have to either hit Esc to get out, or try and fumble around the menu without being able to see anything. Works fine if "Panorama Background On All Screens" is disabled

![2023-07-12_11 54 54](https://github.com/juancarloscp52/BedrockIfy/assets/60888972/08fd82e1-df76-4c00-8c36-0a6b5dabdd01)

**Comment by 7777777-4547** [comment]
1. Add `yacl3.gui.YACLScreen` in `Disable panorama on screens`
![Fix YACL3](https://github.com/juancarloscp52/BedrockIfy/assets/83630775/0ccbb741-ee38-44d5-9ce3-81393d8ae9ad)

2. Restart minecraft

It will maybe fix this issue.

**Comment by lonefelidae16** [comment]
Have you tried adding `yacl3.gui.YACLScreen` into Disable panorama on screens? The namespace of YetAnotherConfigLib seems to have changed since v3.
![2023-07-27_08 18 17](https://github.com/juancarloscp52/BedrockIfy/assets/74746227/2d2604f7-dcf4-46ad-bc07-79a70a04d642)

There is no modset information provided and I don’t have a mod that depends on YACL, so I cannot confirm about this issue.

**Comment by SAJewers** [comment]
Nope, that doesn't work. No change.

**Comment by lonefelidae16** [comment]
Okay I found a solution. This feature have been applied to all tab screens added since 1.20 and a bit of conditions need to be added.

thanks for your time, I’ll create of this fix.

**Comment by jbrown1597** [comment]
How can I locate the "class name" of mods so I may add them to my 'panorama ignore list'?

I'm sticking with this mod, but at the same time it breaks at least half of my config menus.

**Comment by jbrown1597** [comment]
@lonefelidae16 How can I locate the "class name" of mods so I may add them to my "panorama disabled list" & fix broken menus?


Future request: Panorama for loading screen & main menu only, not forced on all possible screens.

THANK YOU FOR ALL YOUR HARD WORK <3

**Comment by lonefelidae16** [comment]
> How can I locate the "class name" of mods so I may add them to my "panorama disabled list" & fix broken menus?

There are two things.

* Open your Jar as a zip and locate .class file. Java code is compiled to Class file and [zipped as Deflate](https://docs.oracle.com/en/java/javase/20/docs/specs/man/jar.html), so you can find the directory for that namespace.
BedrockIfy for instance, me -> juancarloscp52 -> bedrockify -> Bedrockify.class

or

* Decompile it on your IDE. For example on IntelliJ IDEA, click “File” on toolbar -> open “Project Structure” subwindow (<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Shift</kbd>+<kbd>S</kbd>) -> select “Project Settings: Libraries” on the left pane -> add “Java” library -> locate your Jar -> now is placed in “External Libraries” on your project.

> Future request: Panorama for loading screen & main menu only, not forced on all possible screens.

Sorry, but it’s out of this issue, please raise a new one. however I probably can’t solve it right away - will increase setting options.

**Comment by TheChilliPL** [comment]
Any way to have Minecraft e.g. log the current screen name to the console in order to quickly and easily disable them?

**Comment by lonefelidae16** [comment]
> Any way to have Minecraft e.g. log the current screen name to the console in order to quickly and easily disable them?

Yeah there is. I think adding Mixin to the constructor method of the Screen class would be a good idea, but I’m not sure if this way is easy...

```java
public class YourModClass {
    public static final Logger LOGGER = LoggerFactory.getLogger(YourModClass.class);

    // Works on 1.18.2 or later.
    // public static final Logger LOGGER = LogUtils.getLogger();
}
```

```java
@Mixin(Screen.class)
public class ScreenMixin {

    @Inject(method = "<init>(Lnet/minecraft/text/Text;)V", at = @At("RETURN"))
    private void putScreenClassName(Text text, CallbackInfo ci) {
        // Get the current Screen class.
        Screen $this = Screen.class.cast(this);
        // Objective code.
        YourModClass.LOGGER.info("[ScreenLogger] ClassName = '{}'", $this.getClass().getCanonicalName());
    }
}
```

Note that this code generates the line each time when you open new screens, such as opening select world screen, opening inventory, opening options screen, etc.
